File: leetcode/string/0551_Student_Attendance_Record_I/main.py
!/bin/env python3
#coding:utf-8

class Solution:
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        return 'LLL' not in s and s.count('A') <= 1


if __name__ == "__main__":
    assert Solution().checkRecord("PPALLP") == True
    assert Solution().checkRecord("PPALLL") == False
    assert Solution().checkRecord("LALL") == True 
    assert Solution().checkRecord("ALLAPPL") == False

File: leetcode/string/0551_Student_Attendance_Record_I/main.go
package main

import "fmt"

func checkRecord(s string) bool {
    l, a := 0, 0
    for i := 0; i < len(s); i++ {
        if (s[i] == 'L') {
            l += 1
        } else if (s[i] == 'A') {
            l = 0;
            a += 1
        } else {
            l = 0;
        }
        
        if l > 2 || a > 1 {
            return false
        }
    }
    return true
}

func main() {
    fmt.Println(checkRecord("PPALLP"))
}
